services/openshift-service-impl/src/main/java/io/fabric8/launcher/service/openshift/impl/fabric8/openshift/client/ExceptionMapper.java

```diff
@@ -0,0 +1,61 @@
+package io.fabric8.launcher.service.openshift.impl.fabric8.openshift.client;
+
+import java.lang.reflect.InvocationTargetException;
+
+import io.fabric8.kubernetes.client.KubernetesClientException;
+import io.fabric8.launcher.service.openshift.api.DuplicateProjectException;
+import io.fabric8.launcher.service.openshift.api.QuotaExceedException;
+
+/**
+ * Util class that 'maps' KubernetesClientException to domain specific RuntimeException.
+ */
+public class ExceptionMapper {
+
+    /**
+     * For given KubernetesClientException map it to a domain specific runtime Exception,
+     * if none is found return original KubernetesClientException
+     *
+     * @param kce     the KubernetesClientException to find a domain specific alternative for
+     * @param context is used to create the domain specific exception with, be sure it has a constructor of this type
+     * @return the domain specific exception or the original if none is found
+     */
+    public static RuntimeException throwMappedException(KubernetesClientException kce, Object context) {
+        for (ExceptionMapping exceptionMapping : ExceptionMapping.values()) {
+            if (isMatchingException(kce, exceptionMapping)) {
+                return createException(exceptionMapping.exceptionClass, context);
+            }
+        }
+
+        return kce;
+    }
+
+    private static boolean isMatchingException(KubernetesClientException kce, ExceptionMapping exceptionMapping) {
+        return kce.getCode() == exceptionMapping.statusCode
+                && kce.getStatus().getReason().contains(exceptionMapping.statusReason);
+    }
+
+    private static RuntimeException createException(Class<? extends RuntimeException> exceptionClass, Object context) {
+        try {
+            return exceptionClass.getConstructor(context.getClass()).newInstance(context);
```

Comment:
Move this to the enum, and you don't need to use reflection

Reply:
how can you move it to the enum and add the project name without reflection?